src/lib/components/Swap/index.tsx

```diff
@@ -18,54 +23,50 @@ import Settings from './Settings'
 import SwapButton from './SwapButton'
 import Toolbar from './Toolbar'
 
-interface DefaultTokenAmount {
-  address?: string | { [chainId: number]: string }
-  amount?: number
-}
-
-interface SwapDefaults {
-  tokenList: string | TokenInfo[]
-  input: DefaultTokenAmount
-  output: DefaultTokenAmount
-}
-
-const DEFAULT_INPUT: DefaultTokenAmount = { address: 'ETH' }
-const DEFAULT_OUTPUT: DefaultTokenAmount = {}
-
-function useSwapDefaults(defaults: Partial<SwapDefaults> = {}): SwapDefaults {
-  const tokenList = defaults.tokenList || DEFAULT_TOKEN_LIST
-  const input: DefaultTokenAmount = defaults.input || DEFAULT_INPUT
-  const output: DefaultTokenAmount = defaults.output || DEFAULT_OUTPUT
-  input.amount = input.amount || 0
-  output.amount = output.amount || 0
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  return useMemo(() => ({ tokenList, input, output }), [])
-}
-
+export type DefaultAddress = string | { [chainId: number]: string } | 'NATIVE'
 export interface SwapProps {
+  theme?: Theme
+  locale?: SupportedLocale
+  provider?: EthProvider
+  jsonRpcEndpoint?: string
+  width?: string | number
+  dialog?: HTMLElement | null
+  className?: string
+  onError?: ErrorHandler
+  tokenList?: string | TokenInfo[]
+  defaultInputAddress?: DefaultAddress
+  defaultInputAmount?: string
+  defaultOutputAddress?: DefaultAddress
+  defaultOutputAmount?: string
   convenienceFee?: number
-  convenienceFeeRecipient?: string // TODO: improve typing to require recipient when fee is set
-  defaults?: Partial<SwapDefaults>
+  convenienceFeeRecipient?: string | { [chainId: number]: string }
 }
 
-export default function Swap({ defaults }: SwapProps) {
-  const { tokenList } = useSwapDefaults(defaults)
+export default function Swap(props: SwapProps) {
+  const { defaultInputAddress, defaultInputAmount, defaultOutputAddress, defaultOutputAmount, tokenList } = props
+
   useTokenList(tokenList)
+  useSwapDefaults(defaultInputAddress, defaultInputAmount, defaultOutputAddress, defaultOutputAmount)
 
+  const { active, account, chainId } = useActiveWeb3React()
+  const [lastChainId, setLastChainId] = useState<number | undefined>(chainId)
   const [boundary, setBoundary] = useState<HTMLDivElement | null>(null)
-  const { chainId, active, account } = useActiveWeb3React()
 
   // Switch to on-chain currencies if/when chain changes to prevent chain mismatched currencies.
   const [, updateSwapInputCurrency] = useSwapCurrency(Field.INPUT)
   const [, updateSwapOutputCurrency] = useSwapCurrency(Field.OUTPUT)
   const [, updateSwapInputAmount] = useSwapAmount(Field.INPUT)
+
   useLayoutEffect(() => {
-    if (chainId) {
-      updateSwapInputCurrency(nativeOnChain(chainId))
-      updateSwapOutputCurrency()
-      updateSwapInputAmount('')
+    if (chainId !== lastChainId) {
+      setLastChainId(chainId)
```

Comment:
This will reload defaults if the chain experiences a temporary disconnection, which is jarring to a user. Consider having the hook have only `chainId` as a dep, and not storing `lastChainId` in state (see `main` branch for an example).

Reply:
Alternatively, should this logic be a part of `useSwapDefaults`?